app/models/service_ansible_tower.rb

```diff
@@ -17,10 +20,42 @@ def job
     @job ||= service_resources.find { |sr| sr.resource.kind_of?(OrchestrationStack) }.try(:resource)
   end
 
+  def build_stack_options_from_dialog(dialog_options)
+    launch_options = {:extra_vars => extra_vars_from_dialog(dialog_options)}
+    launch_options[:limit] = dialog_options['dialog_limit'] unless dialog_options['dialog_limit'].blank?
+
+    launch_options
+  end
+
   private
 
+  # the method name is required by ServiceOrchestrationOptionMixin
+  def build_stack_create_options
+    # job template from dialog_options overrides the one copied from service_template
+    dialog_options = options[:dialog] || {}
+    if dialog_options['dialog_job_template']
+      self.job_template = ConfigurationScript.find(dialog_options['dialog_job_template'])
+    end
+
+    raise _("job template was not set") if job_template.nil?
+
+    build_stack_options_from_dialog(options[:dialog])
+  end
+
   def save_launch_options
-    # TODO
+    options[:create_options] = dup_and_process_password(job_options)
     save!
   end
+
+  def extra_vars_from_dialog(dialog_options)
+    params = {}
+    dialog_options.each do |attr, val|
+      if attr.start_with?('dialog_param_')
+        params[attr['dialog_param_'.size..-1]] = val
```

Comment:
@bzwei 
Can the 'dialog_param_'.size be a constant instead of being computed in a loop every time